internal/app/server/open_saves.go

```diff
@@ -267,33 +266,16 @@ func (s *openSavesServer) GetRecords(ctx context.Context, req *pb.GetRecordsRequ
 	response := &pb.GetRecordsResponse{
 		Results: []*pb.GetRecordsResponse_Result{},
 	}
-	statusOk := &pb.Status{
-		Code:    uint32(codes.OK),
-		Message: codes.OK.String(),
-	}
 	for i, rec := range records {
-		var protoRec *pb.Record
-		var storeKey string
-		var pbStatus *pb.Status
+		st := status.New(codes.OK, codes.OK.String())
 
-		if rec != nil {
-			protoRec = rec.ToProto()
-			storeKey = rec.StoreKey
-		}
 		if errors != nil && errors[i] != nil {
-			statusCode := status.Code(errors[i])
-			pbStatus = &pb.Status{
-				Code:    uint32(statusCode),
-				Message: errors[i].Error(),
-			}
-		} else {
-			pbStatus = statusOk
+			st = status.Convert(errors[i])
 		}
-
 		result := &pb.GetRecordsResponse_Result{
-			Record:   protoRec,
-			StoreKey: storeKey,
-			Status:   pbStatus,
+			Record:   rec.ToProto(),
```

Comment:
Really nice simplification of the logic with the use of the Status from grpc and changes in the record.Record functions (to test for nil).